src/ApolloHooksMutation.re

```diff
@@ -0,0 +1,267 @@
+type graphqlErrors;
+
+type error = {
+  .
+  "message": string,
+  "graphqlErrors": graphqlErrors,
+};
+
+type refetchQueries =
+  ReasonApolloTypes.executionResult => array(ApolloClient.queryObj);
+
+/* Result that is return by the hook */
+type result('a) =
+  | Data('a)
+  | Error(error)
+  | NoData;
+
+/* Result that is return by the hook */
+type controlledResult('a) = {
+  loading: bool,
+  called: bool,
+  data: option('a),
+  error: option(error),
+};
+
+type controlledVariantResult('a) =
+  | Loading
+  | Called
+  | Data('a)
+  | Error(error)
+  | NoData;
+
+[@bs.module "graphql-tag"] external gql: ReasonApolloTypes.gql = "default";
+
+type mutationResult('a) = {. "data": 'a};
+
+[@bs.deriving abstract]
+type options('a) = {
+  [@bs.optional]
+  variables: Js.Json.t,
+  [@bs.optional]
+  mutation: option(ReasonApolloTypes.queryString),
+  [@bs.optional]
+  client: ApolloClient.generatedApolloClient,
+  [@bs.optional]
+  refetchQueries,
+  [@bs.optional]
+  awaitRefetchQueries: bool,
+  [@bs.optional]
+  update: (ApolloClient.generatedApolloClient, mutationResult('a)) => unit,
+};
+
+type jsResult = {
+  .
+  "data": Js.Nullable.t(Js.Json.t),
+  "loading": bool,
+  "called": bool,
+  "error": Js.Nullable.t(error),
+};
+
+type jsMutate('a) = (. options('a)) => Js.Promise.t(jsResult);
+type mutation('a, 'b) =
+  (
+    ~mutation: 'b,
+    ~client: ApolloClient.generatedApolloClient=?,
+    ~refetchQueries: refetchQueries=?,
+    ~awaitRefetchQueries: bool=?,
+    unit
+  ) =>
+  Js.Promise.t(controlledVariantResult('a));
+
+[@bs.module "@apollo/react-hooks"]
+external useMutationJs:
+  (. ReasonApolloTypes.queryString, options('a)) => (jsMutate('a), jsResult) =
+  "useMutation";
+
+exception Error(string);
+
+let useMutation =
+    (
+      mutation,
```

Comment:
Why here mutation is not labeled? In function which is returned by useDynamicMutation, parameter: mutation is labeled.

Reply:
Following the API of Apollo, and making it more consistent with useQuery. Maybe the for the returned function of `useDynamicMutation` the mutation should also be an unlabeled first argument indeed.

Reply:
See unlabeled mutation in new commit

Reply:
> See unlabeled mutation in new commit
Ok, but in my opinion `useQuery` now should take first argument unlabeled.

Reply:
Agree, didn't realized I hadn't refactored useQuery yet.

Reply:
Changes committed